### PR TITLE
Added Response config to SecurityConfig

### PR DESCRIPTION
**Pull request by karlssonerik**

### CODE REVIEW

**Review by johdah — APPROVED** [review]
Nice, looks good!

File: v2/http-middleware/security.go [hunk 1]
```diff
@@ -224,24 +230,32 @@ func AuthorizeMiddleware(authorizer Authorizer) mux.MiddlewareFunc {
 					WithField("url", req.URL.String()).
 					Error("Couldn't extract User ID from context")
 
+				responseBody := GetInternalServerErrorReponseBody(http_model.ErrResponseInternalServerError, secConfig)
```
Comment (johdah) [inline v2/http-middleware/security.go:233]: Response

**Review by johdah — CHANGES_REQUESTED** [review]

**Review by johdah — APPROVED** [review]

File: v2/http-middleware/security.go [hunk 2]
```diff
@@ -224,24 +230,32 @@ func AuthorizeMiddleware(authorizer Authorizer) mux.MiddlewareFunc {
...
 				return
 			}
 
 			isAuthorized, err := checkAuthorization(ctx, req, authorizer, userID, secConfig.authorizations)
 			var httpErr *http_model.HTTPError
 			if errors.As(err, &httpErr) {
-				http_server.WriteJSONResponse(ctx, w, req, httpErr.StatusCode, httpErr.Message())
+				if secConfig.responses != nil {
+					http_server.WriteJSONResponse(ctx, w, req, http.StatusInternalServerError, secConfig.responses.InternalErrorResponse())
+				} else {
```
Comment (rickardenglund) [inline v2/http-middleware/security.go:246]: this else is not needed?

Comment (karlssonerik) [reply v2/http-middleware/security.go:246]: You mean by adding a return in the if body?

Comment (rickardenglund) [reply v2/http-middleware/security.go:246]: Yes, I just assumed that it was already there ^^

**Pull request merged**